x-pack/plugins/observability/public/pages/slos/components/slo_list_group_by.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+import React, { useState } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiSelectable,
+  EuiSelectableOption,
+  EuiFilterGroup,
+  EuiFilterButton,
+} from '@elastic/eui';
+import { EuiSelectableOptionCheckedType } from '@elastic/eui/src/components/selectable/selectable_option';
+import { SearchState } from '../hooks/use_url_search_state';
+
+export type GroupByField = 'ungrouped' | 'tags'; // TODO add 'sli_indicator' | 'status' | 'instanceId'
+export interface Props {
+  loading: boolean;
+  onChangeGroupBy: (groupBy: GroupByField) => void;
+  initialState: SearchState;
+}
+
+export type Item<T> = EuiSelectableOption & {
+  label: string;
+  type: T;
+  checked?: EuiSelectableOptionCheckedType;
+};
+
+const GROUP_BY_OPTIONS: Array<Item<GroupByField>> = [
+  {
+    label: i18n.translate('xpack.observability.slo.list.groupBy.sliIndicator', {
+      defaultMessage: 'Ungrouped',
+    }),
+    type: 'ungrouped',
+  },
+  {
+    label: i18n.translate('xpack.observability.slo.list.groupBy.tags', {
+      defaultMessage: 'Tags',
+    }),
+    type: 'tags',
+  },
+  // TODO add more options (SLI indicator, status, instance id)
+];
+export function SloGroupBy({ loading, onChangeGroupBy, initialState }: Props) {
+  const [isGroupByPopoverOpen, setGroupByPopoverOpen] = useState(false);
+  const [groupByOptions, setGroupByOptions] = useState<Array<Item<GroupByField>>>(
+    GROUP_BY_OPTIONS.map((option) => ({
+      ...option,
+      checked: option.type === initialState.groupBy ? 'on' : undefined,
+    }))
+  );
+
+  const selectedGroupBy = groupByOptions.find((option) => option.checked === 'on');
+  const handleToggleGroupByButton = () => setGroupByPopoverOpen(!isGroupByPopoverOpen);
+
+  const handleChangeGroupBy = (newOptions: Array<Item<GroupByField>>) => {
+    setGroupByOptions(newOptions);
+    setGroupByPopoverOpen(false);
+    onChangeGroupBy(newOptions.find((o) => o.checked)!.type);
+  };
+  return (
+    <EuiFlexItem grow={false}>
+      <EuiFlexGroup direction="row" gutterSize="s" responsive>
+        <EuiFlexItem style={{ maxWidth: 250 }}>
+          <EuiFilterGroup>
+            <EuiPopover
+              button={
+                <EuiFilterButton
+                  disabled={loading}
+                  iconType="arrowDown"
+                  onClick={handleToggleGroupByButton}
+                  isSelected={isGroupByPopoverOpen}
+                >
+                  {i18n.translate('xpack.observability.slo.list.groupByType', {
+                    defaultMessage: 'Group by {type}',
```

Comment:
🍰 nit: instead of "Group by ungrouped" we could use "none" or we can keep `type: "ungrouped"` but change the text when ungrouped is selected: "No group selected", "none".
Anyway this can be done later.

Reply:
Yep this should be covered [here](https://github.com/elastic/kibana/issues/174331) cc @dominiqueclarke